### PR TITLE
Run multiple tests, reduce disk usage of ubuntu:latest and update som…

### PR DESCRIPTION
**Pull request by burmanm**
…e versions

### CODE REVIEW

**Comment by burmanm** [comment]
Disk usage reduce sudo command from https://github.com/actions/virtual-environments/issues/2606

File: operator/pkg/images/images.go [hunk 1]
```diff
@@ -76,10 +78,11 @@ const (
 )
 
 var imageLookupMap map[Image]string = map[Image]string{
-
-	Cassandra_3_11_6: "datastax/cassandra-mgmtapi-3_11_6:v0.1.5",
```
Comment (jsanda) [inline operator/pkg/images/images.go:80]: Not sure if we should remove this.

File: operator/pkg/images/images.go [hunk 2]
```diff
@@ -171,13 +176,14 @@ func applyDefaultRegistryOverride(image string) string {
 // This is meant to be used when the CassandraDatacenter does not
 // explicitly set the DockerImageRunsAsCassandra field.
 func CalculateDockerImageRunsAsCassandra(version string) bool {
-	if version == "3.11.6" || version == "3.11.7" || version == "4.0.0" {
-		return false
-	}
+	return true
```
[the remark this answers is not in the document]
Comment (burmanm) [reply operator/pkg/images/images.go:179]: Do we want to support those pre-k8ssandra-images to run on k8ssandra-cass-operator? Or should we expect them to just set the parameter? This method only checks if it isn't set (and then figures it out). We have images with 3.11.7 which support and which don't, so version doesn't tell us anything.

File: tests/testdata/additional-service-annotations-and-labels.yaml [hunk 3]
```diff
@@ -17,7 +17,7 @@ spec:
         external-dns.alpha.kubernetes.io/hostname: localhost
   storageConfig:
       cassandraDataVolumeClaimSpec:
-        storageClassName: server-storage
+        storageClassName: standard
```
[the remark this answers is not in the document]
Comment (burmanm) [reply tests/testdata/additional-service-annotations-and-labels.yaml:20]: These are tests, we can modify them any way we want.

[the remark this answers is not in the document]
Comment (burmanm) [reply tests/testdata/additional-service-annotations-and-labels.yaml:20]: Or are you assuming the standard is never WaitForFirstConsume?

[the remark this answers is not in the document]
Comment (burmanm) [reply tests/testdata/additional-service-annotations-and-labels.yaml:20]: But for kind (which we run tests on, it is true): https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/pkg/build/nodeimage/const_storage.go#L129

File: operator/pkg/images/images.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsanda) [reply operator/pkg/images/images.go:179]: If a user wants to migration to the k8ssandra-cass-operator and continue to use the older images, I think it is reasonable for the user to have to explicitly set the `serverImage` property.

[the remark this answers is not in the document]
Comment (burmanm) [reply operator/pkg/images/images.go:179]: But this isn't preventing that?

File: tests/testdata/additional-service-annotations-and-labels.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsanda) [reply tests/testdata/additional-service-annotations-and-labels.yaml:20]: I was assuming standard is never WaitForFirstConsumer and that assumption is wrong 😄

**Review by jsanda — APPROVED** [review]
I am good with the changes. I asked @jimdickinson to do a review as well. It would be nice to get his feedback if possible.

File: operator/pkg/images/images.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsanda) [reply operator/pkg/images/images.go:179]: >But this isn't preventing that?

Sorry, I should have mentioned in my previous comment that I am good with the changes 😄

File: tests/testdata/additional-service-annotations-and-labels.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jimdickinson) [reply tests/testdata/additional-service-annotations-and-labels.yaml:20]: yes - I've yet to see `standard` map to `WaitForFirstConsumer` in a cloud k8s, I think

**Pull request merged**